internal/webhook/v1alpha1/mariadb_webhook.go

```diff
@@ -44,10 +45,14 @@ func (d *MariaDBCustomDefaulter) Default(ctx context.Context, obj runtime.Object
 	}
 	mariadblog.V(1).Info("Defaulting for MariaDB", "name", mariadb.GetName())
 
-	if mariadb.Spec.Replication != nil && mariadb.Spec.Replication.Enabled {
+	env, err := environment.GetOperatorEnv(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting the environment: %v", err)
+	}
+
+	if mariadb.Replication().Enabled {
 		mariadblog.V(1).Info("Defaulting spec.replication", "mariadb", mariadb.Name)
-		mariadb.Spec.Replication.FillWithDefaults()
-		return nil
+		return mariadb.Spec.Replication.SetDefaults(mariadb, env)
```

Comment:
We should completely remove the mutating webhook for defaulting, and rely on the `SetDefaults` method invoked during reconciliation.

Reply:
This also involves removing the `MutatingWebhookConfig` objects from the helm chart and the kubebuilder annotations.